File: backend/internal/models/project.go
package models

import "time"

var ProjectName = "projects"

type Project struct {
	ID        uint      `db:"id" json:"id" goqu:"skipinsert"`
	Name      string    `db:"name" json:"name"`
	UserID    uint      `db:"user_id" json:"user_id"`
	CreatedAt time.Time `db:"created_at" json:"created_at" goqu:"skipupdate"`
	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`
}

File: frontend/src/types/project.ts
import { parseISO } from "date-fns";

export default class Project {
	public id: number;
	public name: string;
	public created_at: Date;
	public updated_at: Date;
	constructor(
		id: string,
		name: string,
		created_at: string,
		updated_at: string
	) {
		this.id = parseInt(id);
		this.name = name;
		this.created_at = parseISO(created_at);
		this.updated_at = parseISO(updated_at);
	}
}
